bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs2/Vcs.scala

```diff
@@ -66,40 +78,48 @@ class Vcs(user: VcsAction[Set[Principal]], server: VcsServer, service: Peer => V
     def filter(eval: MergeEval): Boolean = eval.localUpdate
 
     def update(f: ISPFactory, p: ISPProgram, eval: MergeEval): VcsAction[Unit] =
-      eval.plan.merge(f, p)
+      checkCancel(cancelled) >> eval.plan.merge(f, p)
```

Comment:
For a lot of these things, by the way, the equivalent documentation like the one I quoted is usually clearer in Haskell docs than the scant (if any) doc in scalaz.  For example all you get in scalaz `BindOps` is the method signature.